### PR TITLE
Consistently ignore background class in loss/dice

### PR DESCRIPTION
**Pull request by michaelmckinsey1**

### CODE REVIEW

File: ScaFFold/utils/trainer.py [hunk 1]
```diff
@@ -665,11 +672,11 @@ def train(self):
...
                                 self.spatial_mesh,
                             )
-                            loss_dice = 1.0 - dice_scores.mean()
+                            batch_dice_score = self._foreground_dice_mean(dice_scores)
 
                             # 3. Combine Loss
-                            loss = loss_ce + loss_dice
-                            train_dice_total += dice_scores[:, 1:].mean().item()
+                            loss = loss_ce + (1.0 - batch_dice_score)
+                            train_dice_total += batch_dice_score
```
Comment (PatrickRMiles) [inline ScaFFold/utils/trainer.py:679]: Do we need a `.item()` on `batch_dice_score`? Like `train_dice_total += batch_dice_score.item()`

Comment (michaelmckinsey1) [reply ScaFFold/utils/trainer.py:679]:
```
print(batch_dice_score, type(batch_dice_score))
print(batch_dice_score.item(), type(batch_dice_score.item()))
```
```
DCTensor(0.0005520323175005615, device='cuda:1', grad_fn=<MeanBackward0>) <class 'distconv.distconv.DCTensor'>
0.0005520323175005615 <class 'float'>
```

So apparently it doesn't matter, but to be safe we can convert with `.item()`

**Review by PatrickRMiles — APPROVED** [review]
